### PR TITLE
statistics: add a drop multiple partitions case 

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/48126

Problem Summary:

### What changed and how does it work?

- Refactored drop partition cases use the util to find the event.
- Added tests to cover the reset logic after dropping a partition.
- Added a new test case to cover drop multiple partitions.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

File: pkg/statistics/handle/ddl/ddl_test.go [hunk 1]
```diff
@@ -325,23 +325,108 @@ func TestDropAPartition(t *testing.T) {
...
+	}
+	err = h.Update(is)
+	require.NoError(t, err)
+
+	// Get partition p0 and p1's stats update version.
+	partitionP0ID := pi.Definitions[0].ID
+	partitionP1ID := pi.Definitions[1].ID
+	// Get it from stats_meat first.
+	rows := testKit.MustQuery(
+		"select version from mysql.stats_meta where table_id in (?, ?)",
```
Comment (time-and-fate) [inline pkg/statistics/handle/ddl/ddl_test.go:398]: Add an ORDER BY to make the result stable.

File: pkg/statistics/handle/ddl/ddl_test.go [hunk 2]
```diff
@@ -325,23 +325,108 @@ func TestDropAPartition(t *testing.T) {
...
 	testKit.MustQuery(
-		fmt.Sprintf("select count, modify_count from mysql.stats_meta where table_id = %d", tableInfo.ID),
+		"select count, modify_count from mysql.stats_meta where table_id = ?", tableInfo.ID,
 	).Check(
 		testkit.Rows("3 2"),
 	)
+
+	// Get partition p0's stats update version.
+	partitionID := pi.Definitions[0].ID
+	// Get it from stats_meat first.
```
Comment (time-and-fate) [inline pkg/statistics/handle/ddl/ddl_test.go:342]:
```suggestion
	// Get it from stats_meta first.
```

File: pkg/statistics/handle/ddl/ddl_test.go [hunk 3]
```diff
@@ -325,23 +325,108 @@ func TestDropAPartition(t *testing.T) {
...
+	// the count should be 5 - 2 - 1 = 2 and the modify count should be 2 +1 = 3.
+	testKit.MustQuery(
+		"select count, modify_count from mysql.stats_meta where table_id = ?", tableInfo.ID,
+	).Check(
+		testkit.Rows("2 3"),
+	)
+
+	// Check the update versions are changed.
+	rows = testKit.MustQuery(
+		"select version from mysql.stats_meta where table_id in (?, ?)",
```
Comment (time-and-fate) [inline pkg/statistics/handle/ddl/ddl_test.go:424]: ditto.

File: pkg/statistics/handle/ddl/ddl_test.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/statistics/handle/ddl/ddl_test.go]: Added. Thank you!

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

**Review by time-and-fate — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**